nengo_loihi/tests/test_interneurons.py

```diff
@@ -0,0 +1,72 @@
+import nengo
+import numpy as np
+import pytest
+
+from nengo_loihi.builder import (
+    Model, NoisyInterneurons, OnOffInterneurons,
+    Preset5Interneurons, Preset10Interneurons,
+)
+
+
+@pytest.mark.parametrize('interneurons, tolerance', [
+    (OnOffInterneurons(), 0.32),
+    (NoisyInterneurons(5), 0.11),
+    (NoisyInterneurons(10), 0.10),
+    (Preset5Interneurons(), 0.03),
+    (Preset10Interneurons(), 0.03),
+])
+def test_add_inputs(interneurons, tolerance, Simulator, seed, plt):
+    sim_time = 2.
+    pres_time = 0.5
+    eval_time = 0.25
+
+    stim_fn_a = nengo.processes.Piecewise({
+        t: (2*(i % 2) - 1) * 0.4
+        for i, t in enumerate(np.arange(0, sim_time, pres_time))})
+    stim_fn_b = stim_fn_a
+
+    with nengo.Network(seed=seed) as model:
+        stim_a = nengo.Node(stim_fn_a)
+        stim_b = nengo.Node(stim_fn_b)
+
+        a = nengo.Ensemble(n_neurons=100, dimensions=1)
+        b = nengo.Ensemble(n_neurons=100, dimensions=1)
+
+        nengo.Connection(stim_a, a)
+        nengo.Connection(stim_b, b)
+
+        c = nengo.Ensemble(n_neurons=100, dimensions=1)
+        nengo.Connection(a, c)
+        nengo.Connection(b, c)
+
+        out_synapse = nengo.Alpha(0.03)
+        stim_synapse = out_synapse.combine(
```

Comment:
This feels weird. Why is `stim_synapse` like this?

Reply:
Each time there's a connection in the network, there's two `Lowpass(0.005)` filters applied (one from the ensemble and one from the interneurons). Two identical lowpass filters make an alpha filter, and I combine one of them for each connection. This replicates the filtering and delay that happens in the network, so that the stim signal and recorded output signal are comparable.